conductor/lib/maya_utils.py

```diff
@@ -910,6 +942,106 @@ def is_arnold_tx_enabled():
         return cmds.getAttr("%s.use_existing_tiled_textures" % arnold_node)
 
 
+def scrape_ass_files(ass_filepaths, node_attrs, plugin_paths=()):
+    '''
+    Read/load the given arnold files with the arnold api, and seek out nodes that have filepaths
+    of interest.
+
+    todo(lws): Still need to investigate why this crashes in mayapy when arnold plugins are
+        present (such as yeti or alshaders).
+
+    node_attrs: dictionary. key is the node type, value is a list of node attributes to query.
+
+    plugin_paths: tuple of strings. This may be necessary when running outside of maya/interactive.
+        so that noes for arnold plugins (yeti, etc) can be properly loaded/used.
+
+        e.g. ("/usr/anderslanglands/alshaders/alShaders-linux-2.0.0b2-ai5.0.1.0/bin",
+              "/usr/peregrinelabs/yeti/Maya2017/Yeti-v2.2.6_Maya2017-linux64/bin")
+
+    '''
+    cmds.loadPlugin("mtoa")
+    try:
+        import arnold
+    except ImportError:
+        logger.warning("Failed to import arnold. Could not scrape arnold files: %s", ass_filepaths)
+        return []
+
+    arnold.AiBegin()
+    arnold.AiMsgSetConsoleFlags(arnold.AI_LOG_ALL)
+    if plugin_paths:
+        arnold.AiLoadPlugins(os.path.sep.join(plugin_paths))
+    paths = []
+    for ass_filepath in ass_filepaths:
+        deps = _scrape_ass_file(ass_filepath, node_attrs)
+        paths.extend(deps)
+    arnold.AiEnd()
```

Comment:
Should this be wrapped in a `try-finally` block?

Reply:
good call